flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionConnectionHandlingTest.java

```diff
@@ -131,7 +130,8 @@ private void runTestWithZooKeeperConnectionProblem(
             Problem problem)
             throws Exception {
         CuratorFramework client =
-                ZooKeeperUtils.startCuratorFramework(configuration, NoOpFatalErrorHandler.INSTANCE);
```

Comment:
Before this change, `fatalErrorHandlerResource.getFatalErrorHandler()` was not even used in the test. I don't quite understand why it even caught the exception in the first place. Did it trigger implicitly somewhere?

Reply:
Before, the exception was caught by the `LeaderElectionService` and then passed to the `testingFatalErrorHandler` that failed in the cleanup phase of the test.

Reply:
Ah, thanks. I had seen that `TestingContender` passes errors to the `testingFatalErrorHandler`, but had missed that `LeaderElectionService` has its own `FatalErrorHandler` which just passes the exception on to the contender.